### DISCUSSION TITLE
Interpolation of weather variables for point locations.

### DISCUSSION THREAD

**Discussion by JensvandeZande** [open-meteo/open-meteo#549] [Q&A]
Hi @patrick-zippenfenig ,

I have a question behind the logic of point data. Let's take the historical api as an example. And specifically the era5-land-hourly data. When downloading this data using the CDS API, you need to provide a rectangle for which you want climate data and you will receive data on a grid with a resolution of 0.1° x 0.1°. How did you go about transforming the gridded data from CDS to point location data, since using the historical API of open-meteo you can only specify a point location? Did you use some interpolation techniques or simply used the nearest neighbor? Also, at what resolution did you download the data from era4-land-hourly?

Kind regards,

Jens

**Comment by patrick-zippenfenig** [comment]
Hi, thanks for asking.

I downloaded era5-land for the entire grid (global). Afterwards, I converted it into a time-series for each location (transpose). 

The API picks the nearest grid-cell. However, if the grid-cell is has a different surface elevation than the desired coordinate, it will analyse the surrounding grid-cells and select a more suitable elevation. This drastically improves accuracy in complex terrain. Similar behaviour applies to coastal areas. If you do not want to have this dynamic selection, use `&cell_selection=nearest`.

I do not do any spatial interpolation between grid-cells. Interpolation can reduce accuracy and introduce unwanted artefacts.

**Reply by JensvandeZande** [reply]
Hi @patrick-zippenfenig,

Thanks for your reply! Does the era5-land contain these surface elevation levels or are you referencing a different dataset when you look for more suitable grid cells that have a more appropriate elevation? If so, which dataset?

Also, I have a different question that you can maybe answer me since you are working a lot with this data and I cannot find an answer on the internet. The documentation of era5-land mentions that they capture all these variables at 2 meters above surface level. What does surface level mean? Let's say I want to have data on a hill that is 100 m above sea level, will it return data 2 meters above the hill's surface, so 102 meters above sea level? If I request data for a piece of land that is at sea level it provides me data at 2 m above sea level? How should I interpret this?

Many thanks!

KR Jens

**Comment by mthiboust** [comment]
> I do not do any spatial interpolation between grid-cells. Interpolation can reduce accuracy and introduce unwanted artefacts.

Interesting to know. I though it was a good practice to do spatial interpolation between grid-cells. Could you elaborate on the unwanted artefacts and accuracy reduction?

**Reply by patrick-zippenfenig** [reply]
Grid-cells of a weather model **_should_** represent the average conditions inside a 3x3 km box. The grid-cell center coordinate is the exactly location where data is valid. If you are in the middle of 2 grid-cell centers, I can understand the temptation to interpolate between 2 or more cells. I would never interpolate between weather model grid-cells for 3 reasons:
1. Coastal areas: A grid-cell primarily on sea shows significant different weather conditions and a nearby cell on land. Temperature closer to sea-surface temperature, higher wind speeds due to lower roughness, different low level clouds / fog. If you interpolate between such cells, the result adds no value at all.
2. Complex mountainous terrain: If grid-cells have different altitudes, they have different behaviour for rising/falling air masses that may lead to clouds and precipitation. As processes are physically coupled this will also influence temperature, wind speeds, pressure, etc.
3. Soil type: Each grid-cell assumes different properties for soil type, soil moisture capacity, high/low vegetation, albedo and roughness. I do not know how to interpret interpolated data between a grid-cells assuming forest conditions and a nearby cell assuming more urban conditions.

Instead of interpolation, I would recommend to analyse the terrain, coastal properties and maybe even soil properties of potential grid cells. Open-Meteo is doing this to a certain degree: https://openmeteo.substack.com/p/improving-weather-forecasts-with. 

If you want to use weather data for ML training, it might be a better approach to use data from all surrounding grid-cells and let the ML training figure out which grid-cell fits best or how data from multiple cells can be weighted.

Note: Different datasets like gridded satellite precipitation or solar radiation, that do not have such inherent processes and may benefit from interpolation between grid-cells. Variables like solar radiation or high altitude athmospheric forecasts are also not affected by near surface processes.

**Reply by mthiboust** [reply]
Thank you very much @patrick-zippenfenig for this thorough answer!
